jupyterdrive/gdrive/drive-contents.js

```diff
@@ -96,438 +90,395 @@ define(function(require) {
+         * @return {Promise} A promise fullfilled with the resource of the saved file.
+         */
+        GoogleDriveContents.prototype._upload_new = function (folder_id, model) {
+            var that = this;
+            var converted = contents_model_to_metadata_and_bytes(model);
+            var metadata = converted[0];
+            var contents = converted[1];
+            metadata['parents'] = [{ 'id': folder_id }];
+            if (model['type'] === 'directory') {
+                return gapiutils.execute(gapi.client.drive.files.insert({ 'resource': metadata }));
             }
-        }
-
-        return gapi_utils.gapi_ready
-        .then(function() {
-            return drive_utils.get_id_for_path(path)
-        })
-        .then(function(file_id) {
-            var body = {'title': new_name};
-            var request = gapi.client.drive.files.patch({
-                'fileId': file_id,
-                'resource': body
+            else {
+                return driveutils.upload_to_drive(contents, metadata);
+            }
+        };
+        /**
+         * Notebook Functions
+         */
+        GoogleDriveContents.prototype.get = function (path, options) {
+            var that = this;
+            var metadata_prm = gapiutils.gapi_ready.then($.proxy(driveutils.get_resource_for_path, this, path, 1 /* FILE */));
+            var contents_prm = metadata_prm.then(function (resource) {
+                that._observe_file_resource(resource);
+                return driveutils.get_contents(resource, false);
             });
-            return gapi_utils.execute(request);
-        })
-        .then(function(resource) {
-            that.observe_file_resource(resource);
-            return files_resource_to_contents_model(new_path, resource);
-        });
-    };
-
-    /**
-     * Given a path and a model, save the document.
-     * If the resource has been modifeied on Drive in the
-     * meantime, prompt user for overwrite.
-     **/
-    Contents.prototype.save = function(path, model) {
-        var that = this;
-        var path_and_filename = utils.url_path_split(path);
-        var path = path_and_filename[0];
-        var filename = path_and_filename[1];
-        return drive_utils.get_resource_for_path(path, drive_utils.FileType.FOLDER)
-        .then(function(folder_resource) {
-            return drive_utils.get_resource_for_relative_path(filename, drive_utils.FileType.FILE, false, folder_resource['id'])
-            .then(function(file_resource) {
-                return that.save_existing(file_resource, model)
-            }, function(error) {
-                // If the file does not exist (but the directory does) then a
-                // new file must be uploaded.
-                if (error.name !== 'NotFoundError') {
-                    return Promise.reject(error);
-                }
+            return Promise.all([metadata_prm, contents_prm]).then(function (values) {
```

Comment:
I think it's built-in in typescript now. I don't get any compiler warning. and es6-promise d.ts is not the riht d.ts in some places it fails.